substrate/utils/frame/benchmarking-cli/src/pallet/writer.rs

```diff
@@ -172,6 +176,120 @@ fn map_results(
+	max_extrinsic_weight: Weight,
+	db_weight: RuntimeDbWeight,
+	sanity_weight_check: SanityWeightCheck,
+) -> Result<(), std::io::Error> {
+	if sanity_weight_check == SanityWeightCheck::Ignore {
+		return Ok(())
+	}
+	// Helper function to return max. component value (i.e. max. complexity parameter).
+	fn max_component(parameter: &ComponentSlope, component_ranges: &Vec<ComponentRange>) -> u64 {
+		for component in component_ranges {
+			if parameter.name == component.name {
+				return component.max.into()
+			}
+		}
+		0
+	}
+
+	color_print::cprintln!(
+		"\n<s>Sanity Weight Check 🧐:</> each extrinsic's weight function is executed \
+		in the worst case scenario and compared with the maximum extrinsic weight (the maximum weight \
+		that can be put in a single block for an extrinsic with `DispatchClass::Normal`). In other words, \
+		each extrinsic is checked whether it will fit in an empty (meaning; empty of \
+		`DispatchClass::Normal` extrinsics) block.\n\n<u>Results:</>\n"
+	);
+	let mut sanity_weight_check_passed = true;
+	// Loop through all benchmark results.
+	for ((pallet, instance), results) in all_results.iter() {
+		println!("Pallet: {}\nInstance: {}\n", pallet, instance);
+		for result in results {
+			// Constant `ref_time` & `pov_size`.
+			let mut total_weight = Weight::from_parts(
+				result.base_weight.try_into().unwrap(),
+				result.base_calculated_proof_size.try_into().unwrap(),
+			);
+			// `ref_time` multiplied by complexity parameter.
+			for component in &result.component_weight {
+				total_weight = total_weight.saturating_add(
+					Weight::from_parts(component.slope.try_into().unwrap(), 0)
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// Constant storage reads.
+			total_weight =
+				total_weight.saturating_add(db_weight.reads(result.base_reads.try_into().unwrap()));
+			// Storage reads multiplied by complexity parameter.
+			for component in &result.component_reads {
+				total_weight = total_weight.saturating_add(
+					db_weight
+						.reads(component.slope.try_into().unwrap())
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// Constant storage writes.
+			total_weight = total_weight
+				.saturating_add(db_weight.writes(result.base_writes.try_into().unwrap()));
+			// Storage writes multiplied by complexity parameter.
+			for component in &result.component_writes {
+				total_weight = total_weight.saturating_add(
+					db_weight
+						.writes(component.slope.try_into().unwrap())
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
+			// `pov_size` multiplied by complexity parameter.
+			for component in &result.component_calculated_proof_size {
+				total_weight = total_weight.saturating_add(
+					Weight::from_parts(0, component.slope.try_into().unwrap())
+						.saturating_mul(max_component(&component, &result.component_ranges)),
+				);
+			}
```

Comment:
kinda baffled that all of this math to calculate the final weight of a weight-fn, for a given parameter, needs to be re-created here. Is this really the only place where we have this logic?

Reply:
This logic only exists in the benchmark templates (i.e. [here](https://github.com/paritytech/polkadot-sdk/blob/f50054cffba4df6b25a16dafdbc66d582e419355/substrate/utils/frame/benchmarking-cli/src/pallet/template.hbs#L42-L63) & [here](https://github.com/paritytech/polkadot-sdk/blob/f50054cffba4df6b25a16dafdbc66d582e419355/substrate/.maintain/frame-weight-template.hbs#L57-L76))